flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/triggers/AsyncContinuousEventTimeTrigger.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.windowing.triggers;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.api.common.functions.ReduceFunction;
+import org.apache.flink.api.common.state.v2.ReducingState;
+import org.apache.flink.api.common.state.v2.ReducingStateDescriptor;
+import org.apache.flink.api.common.state.v2.StateFuture;
+import org.apache.flink.api.common.typeutils.base.LongSerializer;
+import org.apache.flink.core.state.StateFutureUtils;
+import org.apache.flink.runtime.asyncprocessing.operators.windowing.triggers.AsyncTrigger;
+import org.apache.flink.streaming.api.windowing.windows.Window;
+
+import java.time.Duration;
+
+/**
+ * A {@link Trigger} that continuously fires based on a given time interval. This fires based on
+ * {@link org.apache.flink.streaming.api.watermark.Watermark Watermarks}.
+ *
+ * @param <W> The type of {@link Window Windows} on which this trigger can operate.
+ * @see org.apache.flink.streaming.api.watermark.Watermark
+ */
+@PublicEvolving
+public class AsyncContinuousEventTimeTrigger<W extends Window> extends AsyncTrigger<Object, W> {
```

Comment:
IIUC, this is not actually used. Can we move this to the package `org.apache.flink.runtime.asyncprocessing.operators.windowing.triggers` and mark as `@Internal`?

Reply:
Resolved